src/corehost/cli/corehost_context_contract.h

```diff
@@ -66,4 +67,15 @@ static_assert(offsetof(corehost_context_contract, run_app) == 5 * sizeof(size_t)
 static_assert(offsetof(corehost_context_contract, get_runtime_delegate) == 6 * sizeof(size_t), "Struct offset breaks backwards compatibility");
 #pragma pack(pop)
 
+/// Signature of delegate returned by coreclr_delegate_type::component_create_native_delegate
+typedef int (__cdecl *ComponentCreateNativeDelegate)(const pal::char_t * assemblyPathNative /* Fully qualified path to assembly */,
+                                                     const pal::char_t * typeNameNative     /* Assembly qualified type name */,
+                                                     const pal::char_t * methodNameNative   /* Public static method name compatible with delegateType */,
+                                                     const pal::char_t * delegateTypeNative /* Assembly qualified delegate type name or null */,
+                                                     int32_t             flags              /* Extensibility flags (currently unused and must be 0) */,
```

Comment:
(Using this PR as the one in coreclr is already closed).
I do agree we need some kind of mechanism for extensibility. That said a single int seems potentially insufficient. I know we've discussed using this to enable unloading eventually, and for that it would probably suffice, but still.
In other places in the native APIs we use a pointer to let us pass additional structure in the future. I would prefer we do that instead. So `const void * unused` for now will work fine. Still need to pass in null.

Reply:
I would agree with this. However, I would suggest `void *` and leave off the `const` unless there is a plan to the make the extensibility point a string type.

Reply:
@AaronRobinsonMSFT is right - it should be `void * unused`.